File: app/src/main/java/org/scp/gymlog/ui/main/routines/RoutineRecyclerViewAdapter.java
package org.scp.gymlog.ui.main.routines;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import org.scp.gymlog.databinding.ListElementFragmentBinding;
import org.scp.gymlog.ui.main.routines.placeholder.PlaceholderContent;
import org.scp.gymlog.ui.main.routines.placeholder.PlaceholderContent.PlaceholderItem;

import java.util.List;

/**
 * {@link RecyclerView.Adapter} that can display a {@link PlaceholderItem}.
 * TODO: Replace the implementation with code for your data type.
 */
public class RoutineRecyclerViewAdapter extends RecyclerView.Adapter<RoutineRecyclerViewAdapter.ViewHolder> {

	private final List<PlaceholderItem> mValues;

	public RoutineRecyclerViewAdapter() {
		mValues = PlaceholderContent.ITEMS;
	}

	@Override
	public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		return new ViewHolder(
				ListElementFragmentBinding.inflate(
						LayoutInflater.from(parent.getContext()), parent, false
				)
		);
	}

	@Override
	public void onBindViewHolder(final ViewHolder holder, int position) {
		holder.mItem = mValues.get(position);
		holder.mContentView.setText(mValues.get(position).content);
	}

	@Override
	public int getItemCount() {
		return mValues.size();
	}

	public class ViewHolder extends RecyclerView.ViewHolder {
		public final TextView mContentView;
		public PlaceholderItem mItem;

		public ViewHolder(ListElementFragmentBinding binding) {
			super(binding.getRoot());
			mContentView = binding.content;
		}

		@Override
		public String toString() {
			return super.toString() + " '" + mContentView.getText() + "'";
		}
	}
}

File: app/src/main/java/org/scp/gymlog/ui/main/routines/RoutineRecyclerViewAdapter.kt
package org.scp.gymlog.ui.main.routines

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import org.scp.gymlog.databinding.ListElementFragmentMuscleBinding
import org.scp.gymlog.ui.main.routines.placeholder.PlaceholderContent

/**
 * [RecyclerView.Adapter] that can display a [PlaceholderItem].
 * TODO: Replace the implementation with code for your data type.
 */
class RoutineRecyclerViewAdapter : RecyclerView.Adapter<RoutineRecyclerViewAdapter.ViewHolder>() {

	private val mValues: List<PlaceholderContent.PlaceholderItem>

	init {
		mValues = PlaceholderContent.ITEMS
	}
	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
		return ViewHolder(
			ListElementFragmentMuscleBinding.inflate(
				LayoutInflater.from(parent.context), parent, false
			)
		)
	}

	override fun onBindViewHolder(holder: ViewHolder, position: Int) {
		holder.mItem = mValues[position]
		holder.mContentView.text = mValues[position].content
	}

	override fun getItemCount(): Int {
		return mValues.size
	}

	inner class ViewHolder(
		binding: ListElementFragmentMuscleBinding
	) : RecyclerView.ViewHolder(binding.root) {

		val mContentView: TextView = binding.content
		var mItem: PlaceholderContent.PlaceholderItem? = null

		override fun toString(): String {
			return super.toString() + " '" + mContentView.text + "'"
		}
	}
}
